diag_manager/fms_diag_outfield.F90

```diff
@@ -0,0 +1,212 @@
+   !! Class fms_diag_outfield_type also contains a significant subset of the fields
+   !! and routines of of the legacy class output_field_type
+   !! TODO: (MDM) This class will need further developemnt for the modern_diag effor.
+   !! For its development, but consider the legacy diag_util::init_output_field already
+   !! in place. Fields added so are uesd the the field buffer math/dupdate functions.
+   !> @ingroup fms_diag_outfield_mod
+   TYPE fms_diag_outfield_type
+      CHARACTER(len=:), ALLOCATABLE :: module_name !< Module name.
+      CHARACTER(len=:), ALLOCATABLE :: field_name  !< Output field name.
+      CHARACTER(len=:), ALLOCATABLE :: output_name !< Output name written to file.
+      CHARACTER(len=:), ALLOCATABLE :: output_file !< File where field should be written.
+
+      !!Major outer loop controls in send_data functions.
+      INTEGER :: pow_value     !< Power value for rms or pow(x) calculations
+      LOGICAL :: phys_window   !< TODO: Rename? OMP subsetted data, See output_fields
+      LOGICAL :: need_compute  !< True iff is local_output and current PE take part in send_data.
+      LOGICAL :: reduced_k_range !< If true, the local start and end indecies are used in k (i.e. 3rd) dim.
+      LOGICAL :: missvalue_present !<
+      LOGICAL :: mask_variant
+      LOGICAL :: mask_present !< True iff mars arguemnt is present in user-facing send function call.
+                              !< Note this field exist since the actual mask argument in the send
+                              !< function call may be downstream replaced by a null pointer which
+                              !< is considered present.
+
+      TYPE(time_reduction_type) :: time_reduction !< Instance of the time_reduction_type.
+
+      !!TODO (Future effort? ) : a pointer for time_min and time_max comparison function
+      !! If possible, this can reove the innermost if/then/else construct in the buffer update loops.
+      !!       min_max_f_ptr => (should point to < or > operators)
+
+      !! gcc error: Interface ‘addwf’ at (1) must be explicit
+      ! procedure (addwf), pointer, nopass :: f_ptr => null () !!A pointer to the field weighing procedure
+
+   CONTAINS
+      procedure, public  :: initialize => initialize_outfield_imp
+   END TYPE fms_diag_outfield_type
+
+
+   !> @brief Class fms_diag_outfield_index_type which (along with class fms_diag_outfield_type)
+   !! encapsulate related information used in updating the output buffers by the diag_manager
+   !! send_data routines. This class in particular focuses on do loop index controls or settings.
+   !! Note that the index names in this class should be indentical to the names used in the
+   !! diag_manager send_data functions and in the "math" buffer update functions. The purpose
+   !! of this class is also to allow for a smaller call function signature for the math/buffer
+   !! update functions.
+   !> @ingroup fms_diag_outfield_mod
+   TYPE, public :: fms_diag_outfield_index_type
```

Comment:
The type name should be` fmsDiagOutFieldIndex_type`

Reply:
Fixed.